ably/realtime_presence_test.go

```diff
@@ -96,3 +104,38 @@ func TestRealtimePresence_Sync250(t *testing.T) {
 		t.Fatalf("members3: %v", err)
 	}
 }
+
+var stateGuard = ^ably.StateEnum(0)
+
+var presTransitions = []ably.StateEnum{
+	ably.StateConnConnecting,
+	ably.StateConnConnected,
+	stateGuard,
+	ably.StateChanAttaching,
+	ably.StateChanAttached,
+}
+
+func TestRealtimePresence_EnsureChannelIsAttached(t *testing.T) {
+	rec := ably.NewStateRecorder()
+	opts := rec.Options()
+	opts.NoConnect = true
+	app, client := testutil.ProvisionRealtime(nil, opts)
+	defer safeclose(t, client, app)
+	channel := client.Channels.Get("persisted:presence_fixtures")
+	if err := ably.Wait(client.Connection.Connect()); err != nil {
+		t.Fatal(err)
+	}
+	rec.Add(stateGuard)
```

Comment:
Event ordering - to ensure I received `StateChanAttached` after calling to `Presence.Get` - if client.Channels.Get automatically attached a channel, the event sequence will be the same, the test would pass but the functionality will be broken - the attach must be initiated here by `Presence.Get`. And inserting marks helps on that :)